core/modules/path/path.module

```diff
@@ -687,19 +687,21 @@ function path_load_multiple_by_entities(array $entities) {
 
   // Each entity may have a different language, so group them by langcode.
   foreach ($entities as $entity) {
-    $uri = $entity->uri();
-    $langcode = isset($entity->langcode) ? $entity->langcode : LANGUAGE_NONE;
-    $sources[$langcode][] = $uri['path'];
-    $map[$langcode][$uri['path']] = $entity->id();
-
-    // Provide a default path in the event no path is found.
-    $entity->path = array(
-      'pid' => NULL,
-      'source' => $uri['path'],
-      'alias' => NULL,
-      'langcode' => $langcode,
-      'auto' => FALSE,
-    );
+    if (!empty($entity->uri)) {
+      $uri = $entity->uri();
```

Comment:
I'm not sure this is correct. Entities like `Node` do not have a `$node->uri` property, but they still return something when calling `$node->uri()`. I think this should be:
```
    $uri = $entity->uri()
    if (!empty($uri)) {
```